src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -121,15 +179,15 @@ public virtual async Task<int> SendJsonRpcResult(JsonRpcResult result)
                 int responseSize = 1;
                 bool isFirst = true;
                 await _stream.WriteAsync(_jsonOpeningBracket);
-                await using JsonRpcBatchResultAsyncEnumerator enumerator = result.BatchedResponses!.GetAsyncEnumerator(CancellationToken.None);
+                await using JsonRpcBatchResultAsyncEnumerator enumerator = result.BatchedResponses!.GetAsyncEnumerator(cancellationToken);
                 while (await enumerator.MoveNextAsync())
                 {
                     JsonRpcResult.Entry entry = enumerator.Current;
                     using (entry)
                     {
                         if (!isFirst)
                         {
-                            await _stream.WriteAsync(_jsonComma);
+                            await _stream.WriteAsync(_jsonComma, cancellationToken);
```

Comment:
Missed `cancellationToken` in line 205

Reply:
IMO `SerializeAsync` should now also take `cancellationToken` (need to add to our json serializer)